src/apy/anki.py

```diff
@@ -34,43 +40,61 @@ def __init__(self, base=None, path=None, profile=None, **_kwargs):
         self.deck_names = self.deck_name_to_id.keys()
         self.n_decks = len(self.deck_names)
 
-    def _init_load_collection(self, base, path, profile):
-        """Load the Anki collection"""
-        # Save CWD (because Anki changes it)
-        save_cwd = os.getcwd()
-
-        if path is None:
-            if base is None:
+    def _init_load_profile(self, base, path):
+        """Load the Anki profile from database"""
+        if base is None:
+            if path is not None:
+                self._path = Path(path)
+            else:
                 click.echo('Base path is not properly set!')
                 raise click.Abort()
 
-            basepath = Path(base)
-            if not (basepath / 'prefs21.db').exists():
-                click.echo('Invalid base path!')
-                click.echo(f'path = {basepath.absolute()}')
-                raise click.Abort()
+        basepath = Path(base)
+        db_path = basepath / 'prefs21.db'
+
+        if not db_path.exists():
+            click.echo('Invalid base path!')
+            click.echo(f'path = {basepath.absolute()}')
+            raise click.Abort()
 
-            # Initialize a profile manager to get an interface to the profile
-            # settings and main database path; also required for syncing
-            self.pm = ProfileManager(base)
-            self.pm.setupMeta()
+        # Load metadata and profiles from database
+        conn = sqlite3.connect(db_path)
+        try:
+            res = conn.execute(
+                "select cast(data as blob) from profiles where name = '_global'"
+            )
+            self._meta = pickle.loads(res.fetchone()[0])
```

Comment:
Right now the `._meta` isn't getting used for anything. I checked out what's inside it, there is a `last_loaded_profile_name` field. This could be a reasonable default if the user doesn't specify a profile in the config. It would behave the same way as the GUI.

Reply:
Thanks, I agree! And as a fallback, we choose the first profile.